lib/takeaway.rb

```diff
@@ -0,0 +1,79 @@
+require "csv"
+require_relative 'dish_list'
+require_relative 'dish'
+require_relative 'order'
+require_relative 'sms'
+
+class Takeaway
+
+  attr_reader :dish_list, :order
+
+  def initialize( args = {dish_list: DishList.new(Dish), order: Order.new} )
+    @dish_list     = args[:dish_list]
+    @order         = args[:order]
+    @placed_orders = []
+  end
+
+  def load_dishes(filename='menu.csv')
+    if File.exist?(filename)
+      CSV.foreach(filename) { |data| dish_list.add_dish(data) }
+    else
+      fail "File '#{filename}' not found!"
+    end
+  end
+
+  def menu
+    puts dish_list
+  end
+
+  def add_dish(id, qty=1)
```

Comment:
This responsibility belongs in `Order`.  What else is `Order` for if not validating dishes and their quantities?